CodeMonkeys.CMS.Public.Shared/Repository/RepositoryBase.cs

```diff
@@ -19,6 +19,28 @@ public RepositoryBase(
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
         }
 
+        protected async Task<TResult?> ExecuteAsync<TResult>(Func<ApplicationDbContext, CancellationToken, Task<TResult>> operation, CancellationToken cancellation = default)
```

Comment:
Trevlig funktion.

Jag är inte säker på om jag ska rekommendera att alla databasoperationer som rimligen kan utföras med den skrivs om för att utföras med den i det här pr:et, men jag är helt säker på att jag rekommenderar att alla databasoperationer som rimligen kan utföras med den skrivs om för att utföras med den.

Inte säker på att explicit hantering av cancellation token på den här nivån är en bra idé. De use cases där funktionen inte gör någonting vettigt med cancellation token känns så dominerande att det skulle kunna vara bättre att hantera de fall där den gör det genom att låta funktionen hämta cancellation token från den miljö där den definieras.

Man kanske skulle kunna flytta in SaveChanges-anropet hit och låta verkställandet av det styras av en boolesk parameter?

Man kanske skulle ha en parameter med en lista med instruktioner för vilka exceptions som ska konverteras till vilka andra exceptions och vilka logmeddelanden som de ska föranleda? (Var på väg att föreslå ett dictionary, men jag är inte säker på om man kan konstruera ett dictionary nycklat på objektklasser och utan vidare hitta den smalaste klass som ett objekt tillhör.)